src/Explorer/Explorer.ts

```diff
@@ -79,6 +79,7 @@ import { UploadItemsPane } from "./Panes/UploadItemsPane";
 import { UploadItemsPaneAdapter } from "./Panes/UploadItemsPaneAdapter";
 import { ReactAdapter } from "../Bindings/ReactBindingHandler";
 import { toRawContentUri, fromContentUri } from "../Utils/GitHubUtils";
+import { ImmutableNotebook } from "@nteract/commutable";

```

Comment:
This needs to be checks for what it does to the JS size. Our JS payload for explorer.js is already massive

Reply:
Btw I created `NotebookManager` which contains all notebook related stuff and is dynamically loaded by explorer. You can put all notebook specific logic in that if needed (to avoid increasing explorer's bundle size).

Reply:
I've removed this import